Mathlib/Analysis/SpecialFunctions/Integrals.lean

```diff
@@ -452,26 +484,51 @@ theorem integral_exp_mul_complex {c : ℂ} (hc : c ≠ 0) :
   · ring
   · fun_prop
 
-@[simp]
-theorem integral_log (h : (0 : ℝ) ∉ [[a, b]]) :
-    ∫ x in a..b, log x = b * log b - a * log a - b + a := by
-  have h' := fun x (hx : x ∈ [[a, b]]) => ne_of_mem_of_not_mem hx h
-  have heq := fun x hx => mul_inv_cancel₀ (h' x hx)
-  convert integral_mul_deriv_eq_deriv_mul (fun x hx => hasDerivAt_log (h' x hx))
-    (fun x _ => hasDerivAt_id x) (continuousOn_inv₀.mono <|
-      subset_compl_singleton_iff.mpr h).intervalIntegrable
-        continuousOn_const.intervalIntegrable using 1 <;>
-  simp [integral_congr heq, mul_comm, ← sub_add]
+/-- Helper lemma for `integral_log`: case where `a = 0` and `b` is positive. -/
+lemma integral_log_from_zero_of_pos (ht : 0 < b) : ∫ s in (0)..b, log s = b * log b - b := by
+  -- Compute the integral by giving a primitive and considering it limit as x
+  -- approaches 0 from the right. The following lines were suggested by Gareth
+  -- Ma on Zulip.
+  rw [integral_eq_sub_of_hasDerivAt_of_tendsto (f := fun x ↦ x * log x - x)
+    (fa := 0) (fb := b * log b - b) (hint := intervalIntegrable_log')]
+  · abel
+  · exact ht
+  · intro s ⟨hs, _ ⟩
+    have h := (hasDerivAt_mul_log hs.ne.symm).sub (hasDerivAt_id s)
+    simp only [id_eq, add_sub_cancel_right] at h
+    exact h
+  · have h := tendsto_log_mul_rpow_nhds_zero zero_lt_one
+    simp_rw [rpow_one, mul_comm] at h
+    replace h := h.sub (tendsto_nhdsWithin_of_tendsto_nhds Filter.tendsto_id)
+    simp only [id_eq, sub_self] at h
+    assumption
+  · apply tendsto_nhdsWithin_of_tendsto_nhds
+    apply ContinuousAt.tendsto
```

Comment:
Can you write something like `exact ContinuousAt.tendsto (by fun_prop)`; or perhaps even combine this with the previous line?

Reply:
Implemented.